src/TextInput.tsx

```diff
@@ -41,18 +49,21 @@ const TextInput = React.forwardRef<HTMLInputElement, TextInputInternalProps>(
       <TextInputWrapper
         block={block}
         className={wrapperClasses}
+        validationStatus={validationStatus}
         contrast={contrast}
         disabled={disabled}
-        hasIcon={!!IconComponent}
+        hasIcon={!!IconComponent || !!(LeadingIconComponent || TrailingIconComponent)}
```

Comment:
Non-blocking nitpick: I prefer `Boolean(isTruthy)` over `!!isTruthy` just because I find it more readable.

Reply:
Removed this check. It was not required after all